angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, OnInit, Input, OnDestroy, HostBinding, Optional, Self, ElementRef, ViewChild } from '@angular/core';
+import { MatFormFieldControl } from '@angular/material';
+import { ControlValueAccessor, NgControl } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { FocusMonitor } from '@angular/cdk/a11y';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import * as moment from 'moment';
+import { firestore } from 'firebase/app';
+
+@Component({
+  selector: 'tanam-date-time',
+  templateUrl: './date-time.component.html',
+  styleUrls: ['./date-time.component.scss'],
+  providers: [
+    {
+      provide: MatFormFieldControl,
+      useExisting: DateTimeComponent,
+    },
+  ],
+})
+export class DateTimeComponent implements MatFormFieldControl<firestore.Timestamp>, ControlValueAccessor, OnInit, OnDestroy {
+
+  private static _nextId = 0;
+
+  @HostBinding('attr.aria-describedby') describedBy = '';
+  @HostBinding() id = `date-time-${DateTimeComponent._nextId++}`;
+  @ViewChild('container', { read: ElementRef }) container: ElementRef;
+
+
+  @Input() type: string;
+  @Input() title: string;
+  @Input() key: string;
+
+
+  stateChanges = new Subject<void>();
+  controlType = 'date-time';
+  shouldLabelFloat = false;
+
+  required: boolean;
+  errorState: boolean;
+  autofilled?: boolean;
+  editorData: any;
```

Comment:
Please change the variable name to `timestamp` and the type should be `firestore.Timestamp`

Reply:
Resolve on this commit https://github.com/oddbit/tanam/pull/56/commits/afe64bb0f806d9ac25dc11b55c4a023e3c778a9d